slider/replay.py

```diff
@@ -152,6 +154,64 @@ def _consume_actions(buffer):
     return out
 
 
+def _ms(t):
+    return t.total_seconds() * 1e3
+
+
+def _within(p1, p2, d):
+    return (p1.x - p2.x) ** 2 + (p1.y - p2.y) ** 2 < d ** 2
+
+
+def _pressed(datum):
+    return datum.key1 or datum.key2 or datum.mouse1 or datum.mouse2
+
+
+def _process_circle(obj, rdatum, hw, scores):
+    out_by = abs(_ms(rdatum.offset - obj.time))
+    if out_by < hw.hit_300:
```

Comment:
yeah i forgot you could do that. I'm kinda reluctant to use timedeltas for things like this because i'm never sure how negatives and the like are handled and i am sure how ints work.